rules/apps.libsonnet

```diff
@@ -59,21 +59,21 @@
             ||| % $._config,
           },
           {
-            record: 'cluster:namespace:pod_memory:active:kube_pod_container_resource_requests',
+            record: 'cluster:namespace:pod_memory:active:kube_pod_resource_requests',
             expr: |||
-              kube_pod_container_resource_requests{resource="memory",%(kubeStateMetricsSelector)s}  * on (namespace, pod, %(clusterLabel)s)
+              kube_pod_resource_requests{resource="memory",%(kubeStateMetricsSelector)s}  * on (namespace, pod, %(clusterLabel)s)
               group_left() max by (namespace, pod, %(clusterLabel)s) (
                 (kube_pod_status_phase{phase=~"Pending|Running"} == 1)
               )
```

Comment:
are we sure that the join on kube_pod_status_phase is still needed?

Reply:
I think we still do, since we probably don't want to estimate the memory of a pod that's in a [`Succeeded`, `Failed`, or `Unknown`](https://github.com/kubernetes/kube-state-metrics/blob/main/internal/store/pod.go#L1296-L1300) state.

Reply:
IIUC the kube-scheduler metrics exclude Succeeded and Failed pods:

https://github.com/kubernetes/kubernetes/blob/45df8f0bb3c0165c5a3e716e803402fa0bd7087d/pkg/scheduler/metrics/resources/resources.go#L186-L201

Reply:
ACK, that just leaves us with pods in the `unknown` state.

Reply:
`Unknown` seems to be [deprecated](https://github.com/kubernetes/kubernetes/blob/b3925d474166b07a7ebeea1e2d81a90183ff5e55/pkg/apis/core/types.go#L2455, `kube_pod_status_phase` is not needed anymore.